e2e/src/snapp.rs

```diff
@@ -66,3 +68,20 @@ pub fn await_state_transition(instance: &SnappAuction, current_state: &[u8]) ->
         .get_current_state_root()
         .wait_and_expect("Could not recover current state root")
 }
+
+pub fn await_and_fetch_auction_bid(instance: &SnappAuction, auction_index: U256) -> AuctionBid {
+    wait_for_condition(|| {
+        instance
+            .auctions(auction_index)
+            .wait_and_expect("No auction bid detected on smart contract")
+            .3
+            != H160::from_str("0000000000000000000000000000000000000000").unwrap()
+    })
+    .expect("Did not detect bid placement in auction");
+
+    AuctionBid::from(
```

Comment:
Nit: You are racing a bit here, since the final query might see a different result than the query inside the `wait_for_condition` closure (what if there is another state change). I guess capturing the result inside the closure mutable will get you in trouble with the borrow checker?

Reply:
You can make the closure mutate captured references, but then `wait_for_condition` needs to accept an `FnMut` instead of `Fn`.

Reply:
Oh my... Looks like I'm in the deep-end now! Im gonna save this part for last.

Reply:
Managed to make this happen with ease (and the help of @fleupold). Thanks for the suggestions fellas!